std/numeric.d

```diff
@@ -171,8 +171,24 @@ struct CustomFloat(uint             precision,  // fraction bits (23 for float)
                    uint             exponentWidth,  // exponent bits (8 for float)  Exponent width
                    CustomFloatFlags flags,
                    uint             bias)
-if (((flags & flags.signed)  + precision + exponentWidth) % 8 == 0 &&
-    precision + exponentWidth > 0)
+    // Restrictions from bitfield
+if ((((flags & flags.signed) + precision + exponentWidth) == 8 ||
+    ((flags & flags.signed) + precision + exponentWidth) == 16 ||
+    ((flags & flags.signed) + precision + exponentWidth) == 32 ||
+    ((flags & flags.signed) + precision + exponentWidth) == 64 ||
+    precision == 64 &&  // due to CustomFloat!80 support hack, see below
+        ((flags & flags.signed) + exponentWidth == 8 ||
+        (flags & flags.signed) + exponentWidth == 16 ||
+        (flags & flags.signed) + exponentWidth == 32 ||
+        (flags & flags.signed) + exponentWidth == 64)) &&
```

Comment:
As bitfields does not allow for more than 64 bits, there is a hack in here, which makes the mantissa a ulong variable, using only the other parts in the bitfield. These lines address this the same way, the tests above did in the standard case.